### ISSUE TITLE
Add `enum_variant_names`

### ISSUE THREAD

**Issue by 0xLucqs** [software-mansion/cairo-lint#14]
# What it does

Detects enumeration variants that are prefixed or suffixed by the same characters.

# Why is this bad?

Enumeration variant names should specify their variant, not repeat the enumeration name.

# Example

```rs
enum Cake {
   BlackForestCake,
   HummingbirdCake,
   BattenbergCake,
}
```

Use instead:

```rs
enum Cake {
   BlackForest,
   Hummingbird,
   Battenberg,
}
```
https://rust-lang.github.io/rust-clippy/master/index.html#/enum_variant_names

**Comment by Ayoazeez26** [comment]
Hi @0xLucqs can I work on this?

**Comment by 0xLucqs** [comment]
yes and please open a PR ASAP so i can track where you are and can help you more easily

**Comment by bloomingpeach** [comment]
Can I work on this issue?

**Comment by 0xLucqs** [comment]
yes

**Comment by MariangelaNM** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/cairo-lint).

### My background and how it can be leveraged
Hi @0xLucqs can I work on this?

**Comment by adpthegreat** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/cairo-lint).

### My background and how it can be leveraged
I am a rust developer of close to a year that is interested in open source and tackling problems   I recently made contributions to the technical content of solana-developers and I aim looking to make more meaningful contributions to other rust based projects in the web3 ecosystem 

### How I plan on tackling this issue
I would first read through the codebase and look at how other lints were implemented to have an idea of how I can model my solution , then create the file for the lint and write the appropriate logic that matches the lint rules

**Comment by SumitGoulikar** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/cairo-lint).

### My background and how it can be leveraged
I'm an undergraduate in Computer Science and am well experienced in coding across different programming languages. Along the way, I have enjoyed building efficient algorithms, practicing data structures, and tackling problems in clean, optimized code. So far, I've had fun using best practices to enhance performance and readability in projects.

I'm excited about this project because it will take my coding abilities, problem-solving skills, and passion for learning to the benefit of improving and maintaining code quality. With my strong foundation in computer science and eagerness to collaborate on impactful projects, you can expect to receive valuable contributions from me.
